### DISCUSSION TITLE
Command and Options to Get Nice Command-Line Search Output?

### DISCUSSION THREAD

**Discussion by RandomDSdevel** [Homebrew/discussions#2523] [Tap maintenance and Homebrew development] [answered]
@XuehaiPan:  

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;One quick question; what did you use to get the nice command-line search output you had in https://github.com/Homebrew/brew/pull/12465#discussion_r754866984?  I want something similar for something else.  Either it wasn't '`grep`' or I can't figure out the particular combination of flags you passed it.

**Comment by XuehaiPan** [comment, marked as answer]
> One quick question; what did you use to get the nice command-line search output you had in https://github.com/Homebrew/brew/pull/12465#discussion_r754866984?

@RandomDSdevel I do not quite understand the question. Did you mean the search result in the other comment https://github.com/Homebrew/brew/pull/12465#discussion_r755067904? I will answer this if you meant that.

I'm using `vscode`'s built-in search panel with regex on:

![Screenshot vscode](https://user-images.githubusercontent.com/16078332/143186393-a6be9323-f17b-43dd-a67e-db40ae31a060.png)

For command line tool, you can get similar results using [`ripgrep`](https://github.com/BurntSushi/ripgrep) (command is `rg`) with `--context=2`:

```console
$ brew install ripgrep

$ rg --context=2 'system HOMEBREW_BREW_FILE, "(install|upgrade)"'
Library/Homebrew/dev-cmd/bottle.rb
112-
113-      ohai "Installing #{f}..."
114:      safe_system HOMEBREW_BREW_FILE, "install", f
115-    end
116-  end
--
266-    unless gnu_tar.any_version_installed?
267-      ohai "Installing `gnu-tar` for bottling..."
268:      safe_system HOMEBREW_BREW_FILE, "install", "--formula", gnu_tar.full_name
269-    end
270-

Library/Homebrew/dev-cmd/bump.rb
61-        unless Pathname.new(ENV["HOMEBREW_BREWED_CURL_PATH"]).exist?
62-          ohai "Installing `curl` for Repology queries..."
63:          safe_system HOMEBREW_BREW_FILE, "install", "--formula", Formula["curl"].full_name
64-        end
65-      rescue FormulaUnavailableError

Library/Homebrew/dev-cmd/tests.rb
52-      ohai "Installing `buildpulse-test-reporter` for reporting test flakiness..."
53-      with_env(HOMEBREW_NO_AUTO_UPDATE: "1", HOMEBREW_NO_BOOTSNAP: "1") do
54:        safe_system HOMEBREW_BREW_FILE, "install", "buildpulse-test-reporter"
55-      end
56-    end
...
```

![Screenshot ripgrep](https://user-images.githubusercontent.com/16078332/143187040-fa277f78-a7fd-4c76-8d9a-8121d6f9d00b.png)

**Comment by carlocab** [comment]
> For command line tool, you can get similar results using [`ripgrep`](https://github.com/BurntSushi/ripgrep) (command is `rg`) with `--context=2`

There's also the `-C` short flag.
```
❯ rg -C2 'system HOMEBREW_BREW_FILE, "(install|upgrade)"' "$(brew --repo)"
/usr/local/Homebrew/Library/Homebrew/utils.rb
466-    unless formula.any_version_installed?
467-      ohai "Installing `#{formula.name}`#{reason}..."
468:      safe_system HOMEBREW_BREW_FILE, "install", "--formula", formula.full_name
469-    end
470-
471-    if latest && !formula.latest_version_installed?
472-      ohai "Upgrading `#{formula.name}`#{reason}..."
473:      safe_system HOMEBREW_BREW_FILE, "upgrade", "--formula", formula.full_name
474-    end
475-

/usr/local/Homebrew/Library/Homebrew/cmd/update-report.rb
456-
457-        ohai "Installing #{new_name}..."
458:        system HOMEBREW_BREW_FILE, "install", new_full_name
459-        begin
460-          unless Formulary.factory(new_full_name).keg_only?
--
482-          system HOMEBREW_BREW_FILE, "cleanup"
483-          ohai "brew install --cask #{new_name}"
484:          system HOMEBREW_BREW_FILE, "install", "--cask", new_name
485-          ohai <<~EOS
486-            #{name} has been moved to Homebrew Cask.
```
